docs/logging.md

```diff
@@ -0,0 +1,75 @@
+# Logging
+
+Thanos supports query logging via flags.
```

Comment:
I think we can make this statement even more generic, since this mechanism supports _request_ logging in general, which means it works for all components.